cdap-docs/developers-manual/source/building-blocks/index.rst

```diff
@@ -48,6 +49,11 @@ write data through the data abstraction layer in CDAP.
 - :doc:`Workers <workers>`
 - :doc:`Workflows <workflows>`
 
+:doc:`Metadata <metadata>` |---| consisting of **properties** (a list of key-value pairs)
+or **tags** (a list of keys) |---| can be set for both data and applications. These can be
```

Comment:
Do we have a notion of data that includes datasets and streams in the documentation already? Other wise it would be better to callout datasets, streams, applications and programs separately here.